### PR TITLE
chore(babel): added babel to support es6

### PR DESCRIPTION
**Pull request by EladBezalel**

### CODE REVIEW

**Comment by Frank3K** [comment]
What about [babel-preset-env](https://github.com/babel/babel-preset-env)? It seems like that is [the recommendation](https://twitter.com/left_pad/status/837443603379204096) now.

**Comment by EladBezalel** [comment]
@Frank3K I certainly thought of it but i realized it shouldn't be env based..
But now i'm thinking of it again and maybe we should do 
```
["env", {
      "targets": {
        "browsers": ["last 2 versions", "safari >= 7"]
      }
    }]
```

**Comment by Frank3K** [comment]
I think you should use the list that is used by autoprefixer (in scripts/gulp-utils.js).

**Comment by EladBezalel** [comment]
@DevVersion what do you think?

**Comment by DevVersion** [comment]
@EladBezalel I think that the `babel-preset-env` would work. Also like the idea of sharing the browsers array with the one from the autoprefixer.

**Comment by Frank3K** [comment]
Note that you can add the browserlist to package.json, or set it in a separate config file as described [here](https://github.com/ai/browserslist/blob/master/README.md). Autoprefixer and babel-env-preset will automatically pick it up.

**Comment by EladBezalel** [comment]
@Frank3K @DevVersion please review

File: gulp/util.js [hunk 1]
```diff
@@ -48,6 +49,9 @@ function buildJs () {
 
   var jsBuildStream = gulp.src( jsFiles )
       .pipe(filterNonCodeFiles())
+      .pipe(babel({
+          presets: [["env", {targets: {browsers: config.browsers}}]]
```
Comment (Frank3K) [inline gulp/util.js]: I think `uglify: true` should be added here. [Readme](https://github.com/babel/babel-preset-env#targetsuglify) and [PR](https://github.com/babel/babel-preset-env/pull/178).

Comment (EladBezalel) [inline gulp/util.js]: sounds like `useBuiltIns` would work better

**Comment by ThomasBurleson** [comment]
This is look good. Let's finish with final changes and one last review.

File: gulp/util.js (hunk 1 above)
Comment (Frank3K) [inline gulp/util.js]: If I understand it correctly `useBuiltIns` alone won't solve the minification issues. I'm not sure what uglify + `useBuiltIns` does.

The option `useBuiltIns`seems to require a module loader (see [explanation] (https://github.com/babel/babel-preset-env/releases/tag/v1.0.0)).

**Comment by ThomasBurleson** [comment]
Closing as not valid for stabilization.

**Pull request closed without merging**